input/pagecontent/reference.md

```diff
@@ -33,33 +116,40 @@ The primitive data types and value formats in FSH are identical to the [primitiv
 
-#### References to External FHIR Artifacts
+#### Referring to Other Items
 
-FHIR resources, profiles, extensions, and value sets defined outside the [FSH tank](index.html#fsh-files-and-fsh-tanks) are referred to by their canonical URIs. Base FHIR resources can also be referred to by their id, for example, `Patient` or `Observation`. In cases where an IG defines a profile or extension matching an existing FHIR ID, use the canonical URL to refer to the FHIR resource.
+FSH items within the same project can be referred to by their names or ids, preferably, as given in their [declaration statement](#defining-items).
 
-#### Files
+External FHIR artifacts in FHIR core and external IGs can be referred to by name, id, or canonical URL. Referring to core FHIR resources by name, e.g., `Patient` or `Observation`, is recommended. For other external items, the use of canonical URLs is recommended, since this approach minimizes the chance of name collisions. In cases where an external name or id clashes with an internal name or id, then the internal entity takes precedence, and external entity must be referred to by its canonical URL.
+
+#### Reference and Canonical Data Types
 
-Content in one project can be contained in one or more files with **.fsh** extension. How content is divided among files is not meaningful in FSH, and all content can be considered pooled together for the purposes of FSH. It is up to implementations to define which **.fsh** files should be included in a given project.
+FHIR resources contain [two types of references](https://www.hl7.org/fhir/references.html) to other resources:
 
-#### Order of Items in Files
+* Resource references
+* Canonical references
 
-The items defined by FSH are: Aliases, Profiles, Extensions, Instances, Value Sets, Code Systems, Mappings, Rule Sets, and Invariants. Items can appear in any order within **.fsh** files, and can be moved around within a file (or to other **.fsh** files) without affecting the interpretation of the content.
+FSH represents Resource references using the syntax `Reference({Resource  name|id|url})`. Where the type is Reference, `Reference()` is required.
+
+Canonical references refer to the standard URL associated with a type of FHIR resource. For elements that require a canonical reference, FSH will accept a URL, or an expression in the form `Canonical({name|id})` where `name` and `id` refer to items defined in the same FSH project. Implementations are to interpret `Canonical()` as an instruction to construct the canonical URL for the referenced item using the FSH project's canonical URL. `Canonical()` therefore enables a user to change the project’s canonical URL in a single place with no changes to FSH definitions.
```

Comment:
Again, I would make this a SUSHI *bonus feature*, not a part of the spec.

Reply:
Why not have it in the spec?  There are places where it might be useful...  And from our implementation standpoint, it was actually easier to support it for _everything_ than it would have been if we thought we had to support it _only_ for FSH items.  The restriction to FSH items only may actually cause more burden on implementors, even thought it might seem it should be less.

Reply:
Consistency ("this works everywhere") is also easier on authors than special rules ("this works only here").  And since `Canonical` has some parallels to `Reference`, it's also probably easier on authors if they work on the same sets of items.